### PR TITLE
Pass options through to LevelDB driver

### PR DESCRIPTION
**Pull request by dizzyd**

### CODE REVIEW

File: src/riak_kv_eleveldb_backend.erl [hunk 1]
```diff
@@ -310,6 +297,32 @@ callback(_Ref, _Msg, State) ->
...
+open_db(DataRoot, Config) ->
+    %% Get the data root directory
+    filelib:ensure_dir(filename:join(DataRoot, "dummy")),
+
+    %% Use a variable write buffer size in order to reduce the number
+    %% of vnodes that try to kick off compaction at the same time
+    %% under heavy uniform load...
+    WriteBufferMin = config_value(write_buffer_size_min, Config, 3 * 1024 * 1024),
+    WriteBufferMax = config_value(write_buffer_size_max, Config, 6 * 1024 * 1024),
+    random:seed(now()),
```
Comment (jonmeredith) [inline src/riak_kv_eleveldb_backend.erl]: Why do you re-seed after L78?

File: src/riak_kv_eleveldb_backend.erl [hunk 2]
```diff
@@ -73,30 +75,19 @@ api_version() ->
...
-    case eleveldb:open(DataDir, Options) of
+    case open_db(DataDir, Config) of
         {ok, Ref} ->
-            {ok, #state {ref = Ref,
-                         data_root = DataDir,
-                         write_buffer_size = WriteBufferSize }};
+            {ok, #state { ref = Ref,
+                          data_root = DataDir,
+                          read_opts = config_value(read_options, Config, []),
+                          write_opts = config_value(write_options, Config, []),
```
Comment (jonmeredith) [inline src/riak_kv_eleveldb_backend.erl:89]: Should fold_opts be configurable from Config too?

**Comment by kellymclaughlin** [comment]
Looks good. +1 to merge.

File: src/riak_kv_eleveldb_backend.erl (hunk 2 above)
Comment (dizzyd) [inline src/riak_kv_eleveldb_backend.erl:89]: Fixed.

File: src/riak_kv_eleveldb_backend.erl (hunk 1 above)
Comment (dizzyd) [inline src/riak_kv_eleveldb_backend.erl]: Fixed.

**Pull request merged**